crates/generator/src/account_whitelist.rs

```diff
@@ -0,0 +1,53 @@
+use gw_common::H256;
+use gw_types::offchain::RunResult;
+use log::debug;
+
+pub struct SUDTProxyAccountWhitelist {
+    allowed_sudt_proxy_creator_account_id: Vec<u32>,
+    sudt_proxy_code_hashes: Vec<H256>,
+}
+
+impl SUDTProxyAccountWhitelist {
+    pub fn new(
+        allowed_sudt_proxy_creator_account_id: Vec<u32>,
+        sudt_proxy_code_hashes: Vec<H256>,
+    ) -> Self {
+        Self {
+            allowed_sudt_proxy_creator_account_id,
+            sudt_proxy_code_hashes,
+        }
+    }
+
+    /// Only accounts in white list could create sUDT proxy contract.
+    pub fn validate(&self, run_result: &RunResult, from_id: u32) -> bool {
+        for k in run_result.write_data.keys() {
```

Comment:
We have three shortcuts:

1. if from_id in the whitelist, we could immediately return true.
2. if run_result.scripts.is_empty, we could also immediately return true since there are no new account been created.
3. if the to account's script.code_hash isn't polyjuice, we can return true.

otherwise we should read code_hash from state tree, instead of iterate all keys.

Reply:
Please rename this module to `erc20_creator_whitelist.rs`